shell/platform/embedder/embedder.h

```diff
@@ -0,0 +1,102 @@
+// Copyright 2017 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+#ifndef FLUTTER_EXPORT
+#define FLUTTER_EXPORT
+#endif  // FLUTTER_EXPORT
+
+#define FLUTTER_ENGINE_VERSION 1
+
+typedef enum {
+  kSuccess = 0,
+  kInvalidLibraryVersion,
+  kInvalidArguments,
+} FlutterResult;
+
+typedef enum {
+  kOpenGL,
+} FlutterRendererType;
+
+typedef void* FlutterEngine;
+
+typedef bool (*BoolCallback)(void* /* user data */);
+typedef uint32_t (*UIntCallback)(void* /* user data */);
+
+typedef struct {
+  BoolCallback make_current;
+  BoolCallback clear_current;
+  BoolCallback present;
+  UIntCallback fbo_callback;
+} FlutterOpenGLRendererConfig;
+
+typedef struct {
+  FlutterRendererType type;
+  union {
+    FlutterOpenGLRendererConfig open_gl;
+  };
+} FlutterRendererConfig;
+
+typedef enum {
+  kWindowMetrics,
+  kPointer,
+} FlutterEventType;
+
+typedef struct {
+  size_t width;
+  size_t height;
+} FlutterWindowMetricsEvent;
+
+typedef enum {
+  kCancel,
+  kUp,
+  kDown,
+  kMove,
+} FlutterPointerPhase;
+
+typedef struct {
```

Comment:
Consider the implementation of FlutterEngineSendPointerEvent.  It receives a point to some memory.  How many bytes from the start of the point can the implementation of FlutterEngineSendPointerEvent read?  The answer depends on the sizeof(FlutterEngineSendPointerEvent) that the client knew when it was compiled.

If we add fields to the struct, the implementation doesn't know whether the client was compiled before or after the fields were added.  There are a bunch of solutions to this problem, but IMHO the easiest is to have the client store sizeof(FlutterEngineSendPointerEvent) as the first field of the struct.  Then the implementation of FlutterEngineSendPointerEvent knows how many bytes it can read and consequently which fields the client included.